quantum/encoder.h

```diff
@@ -22,45 +22,76 @@
 #include "gpio.h"
 #include "util.h"
 
+#ifdef ENCODER_ENABLE
+
+__attribute__((weak)) bool should_process_encoder(void);
+
 void encoder_init(void);
-bool encoder_read(void);
+bool encoder_task(void);
+bool encoder_queue_event(uint8_t index, bool clockwise);
 
 bool encoder_update_kb(uint8_t index, bool clockwise);
 bool encoder_update_user(uint8_t index, bool clockwise);
 
-#ifdef SPLIT_KEYBOARD
+#    ifdef SPLIT_KEYBOARD
```

Comment:
This block is still making the assumption that pins are defined for all of the encoders. Which may not be the case, and essentially disables the encoder functionality, if pins aren't set.

Reply:
Right now we don't have a better way of doing this, and we currently allow for `NO_PIN` to be specified in either info.json or config.h

I'd suggest we can work out another mechanism for handling a different number of encoders per side without being specifically referred to by pins, but I'd like to keep that to a followup PR.